gear-test/src/sample.rs

```diff
@@ -95,9 +102,10 @@ pub struct Expectation {
     /// Flag, which points that errors are allowed. Could be used to check traps.
     #[serde(rename = "allowError")]
     pub allow_error: Option<bool>,
+    /// If this flag turned on, we check that only tested programs was executed
+    pub exact: Option<bool>,
```

Comment:
Rename it to `exact_programs` for clarification. But I would like to see something like this in yamls:

NOT LIKE THIS
```yaml
    expected:
       - exact_programs: true

       # no active programs in the state
       - programs:
         - kind: id
           value: 1
           terminated: true
```

BUT LIKE THIS
```yaml
    expected:
       # no active programs in the state
       - programs:
         only: true
         - kind: id
           value: 1
           terminated: true
```

it might be done with `programs` represented by
```
struct ProgramParse {
    only: Option<bool>,
    #[flatten]
    progs: Option<Vec<ChainProgram>>
}
```

Reply:
I tried yours way, but get 'Error decoding '...gear/gear-test/spec/test_create_program.yaml': mapping values are not allowed in this context at line 30 column 17'
```

Reply:
done